do not ignore errors on `make -j uninstall` and `make distclean`

the flows for `hwloc`, `mpix`, `openmpi` and `slurm` run `make -j uninstall` and `make distclean` before `make install`, and to silence errors if those are not installed, use `ignore_errors: yes`. This practice is too inclusive, as it can also hide more than just `not targets` errors.
to mitigate such errors, i recommend using this instead:
```
  failed_when:
    - rc != 0
    - "\"make: *** No rule to make target 'distclean'\" not in stderr"
  changed_when:
    - rc == 0
    - "\"rm -f\" in stdout"
```

PR is coming.

--- a/roles/openmpi/tasks/main.yml
+++ b/roles/openmpi/tasks/main.yml
@@ -89,7 +89,13 @@
   command: make -j uninstall
   args:
     chdir: "{{ openmpi_build_dir }}"
-  ignore_errors: yes
+  register: openmpi_make_uninstall
+  failed_when:
+    - openmpi_make_uninstall.rc != 0
+    - "\"make: *** No rule to make target 'uninstall'\" not in openmpi_make_uninstall.stderr"
+  changed_when:
+    - openmpi_make_uninstall.rc == 0
+    - "\"uninstall: rm -f\" in openmpi_make_uninstall.stdout"
   when: openmpi_build
   tags:
     - uninstall
@@ -98,7 +104,13 @@
   command: make distclean
   args:
     chdir: "{{ openmpi_build_dir }}"
-  ignore_errors: yes
+  register: openmpi_make_distclean
+  failed_when:
+    - openmpi_make_distclean.rc != 0
+    - "\"make: *** No rule to make target 'distclean'\" not in openmpi_make_distclean.stderr"
+  changed_when:
+    - openmpi_make_distclean.rc == 0
+    - "\"rm -f\" in openmpi_make_distclean.stdout"
   when: openmpi_build
 
 - name: configure openmpi

--- a/roles/slurm/tasks/build.yml
+++ b/roles/slurm/tasks/build.yml
@@ -124,7 +124,13 @@
   command: make -j uninstall
   args:
     chdir: "{{ slurm_build_dir }}"
-  ignore_errors: yes
+  register: slurm_make_uninstall
+  failed_when:
+    - slurm_make_uninstall.rc != 0
+    - "\"make: *** No rule to make target 'uninstall'\" not in slurm_make_uninstall.stderr"
+  changed_when:
+    - slurm_make_uninstall.rc == 0
+    - "\"uninstall: rm -f\" in slurm_make_uninstall.stdout"
   when: slurm_build
   tags:
     - uninstall
@@ -133,7 +139,13 @@
   command: make distclean
   args:
     chdir: "{{ slurm_build_dir }}"
-  ignore_errors: yes
+  register: slurm_make_distclean
+  failed_when:
+    - slurm_make_distclean.rc != 0
+    - "\"make: *** No rule to make target 'distclean'\" not in slurm_make_distclean.stderr"
+  changed_when:
+    - slurm_make_distclean.rc == 0
+    - "\"rm -f\" in slurm_make_distclean.stdout"
   when: slurm_build
 
 - name: configure

--- a/roles/slurm/tasks/hwloc.yml
+++ b/roles/slurm/tasks/hwloc.yml
@@ -67,7 +67,13 @@
   command: make -j uninstall
   args:
     chdir: "{{ hwloc_build_dir }}"
-  ignore_errors: yes
+  register: hwloc_make_uninstall
+  failed_when:
+    - hwloc_make_uninstall.rc != 0
+    - "\"make: *** No rule to make target 'uninstall'\" not in hwloc_make_uninstall.stderr"
+  changed_when:
+    - hwloc_make_uninstall.rc == 0
+    - "\"uninstall: rm -f\" in hwloc_make_uninstall.stdout"
   when: hwloc_build
   tags:
     - uninstall
@@ -76,7 +82,13 @@
   command: make distclean
   args:
     chdir: "{{ hwloc_build_dir }}"
-  ignore_errors: yes
+  register: hwloc_make_distclean
+  failed_when:
+    - hwloc_make_distclean.rc != 0
+    - "\"make: *** No rule to make target 'distclean'\" not in hwloc_make_distclean.stderr"
+  changed_when:
+    - hwloc_make_distclean.rc == 0
+    - "\"rm -f\" in hwloc_make_distclean.stdout"
   when: hwloc_build
 
 - name: configure hwloc

--- a/roles/slurm/tasks/pmix.yml
+++ b/roles/slurm/tasks/pmix.yml
@@ -68,7 +68,13 @@
   command: make -j uninstall
   args:
     chdir: "{{ pmix_build_dir }}"
-  ignore_errors: yes
+  register: pmix_make_uninstall
+  failed_when:
+    - pmix_make_uninstall.rc != 0
+    - "\"make: *** No rule to make target 'uninstall'\" not in pmix_make_uninstall.stderr"
+  changed_when:
+    - pmix_make_uninstall.rc == 0
+    - "\"uninstall: rm -f\" in pmix_make_uninstall.stdout"
   when: pmix_build
   tags:
     - uninstall
@@ -77,7 +83,13 @@
   command: make distclean
   args:
     chdir: "{{ pmix_build_dir }}"
-  ignore_errors: yes
+  register: pmix_make_distclean
+  failed_when:
+    - pmix_make_distclean.rc != 0
+    - "\"make: *** No rule to make target 'distclean'\" not in pmix_make_distclean.stderr"
+  changed_when:
+    - pmix_make_distclean.rc == 0
+    - "\"rm -f\" in pmix_make_distclean.stdout"
   when: pmix_build
 
 - name: configure pmix

